### ISSUE TITLE
[OpenMP][Flang] Missed semantic check for simd aligned clause

### ISSUE THREAD

**Issue by DominikAdamski** [llvm/llvm-project#61161]
OpenMP 5.2 standard states that only pointer variables (`C_PTR`, Cray pointers, `POINTER` or `ALLOCATABLE` items) can appear in SIMD aligned clause (section 5.11). LLVM Flang does not check this restriction. In consequence LLVM Flang does not generate error message for such code:

```f90
subroutine simdloop_aligned(n)
  integer :: i, n
  integer :: A(n)
  !$OMP SIMD ALIGNED(A:256)
  do i = 1, n
    A(i) = i
  end do
end subroutine
```
Compilation command:
```shell
bin/bbc -fopenmp -emit-fir simd.f90
```

LLVM/Flang version: ae9288d471ec02c8812091ba43b92a5dbec6c38f

**Comment by tblah** [comment]
Compiling this with current flang main I new get
```
error: 'a' in ALIGNED clause must be of type C_PTR, POINTER, or ALLOCATABLE
```
